index.js

```diff
@@ -1,11 +1,134 @@
 const core = require('@actions/core');
+const github = require('@actions/github')
+const semver = require('semver');
+const { getConfig } = require('./config');
+const { extractPRNumber, fetchPR, getReleaseType, getReleaseNotes } = require("./pr");
+const { octokit } = require("./octokit");
 
 async function run() {
     try {
-        core.info(`Hello, World!`);
-    } catch (error) {
-        core.setFailed(error.message);
+        var config = getConfig()
+        if (config.mode === 'validate') {
+            await validateActivePR(config)
+        } else if (config.mode === 'bump') {
+            await bumpAndTagNewVersion(config)
+        }
+    } catch (e) {
+        core.setFailed(`unexpected error: ${e.message}`)
     }
 }
 
+// Ensures that the currently active PR contains the required release metadata.
+async function validateActivePR(config) {
+    if (!isActivePR()) {
+        core.warning(`in 'validate' mode, but this doesn't look like an active PR event (is your workflow misconfigured?)`)
+        return
+    }
+
+    var pr
+    try {
+        pr = await fetchPR(github.context.payload.pull_request.number)
+    } catch (e) {
+        core.setFailed(e.message)
+        return
+    }
+
+    var releaseType, releaseNotes
+    try {
+        releaseType = getReleaseType(pr, config.releaseLabels)
+        releaseNotes = getReleaseNotes(pr, config.releaseNotesRegex, config.requireReleaseNotes)
+    } catch (e) {
+        core.setFailed(`PR validation failed: ${e.message}`)
+        return
+    }
+
+    const currentVersion = await getCurrentVersion()
+    const newVersion = semver.inc(currentVersion, releaseType)
+
+    core.info(`current version: ${config.v}${currentVersion}`)
+    core.info(`next version: ${config.v}${newVersion}`)
+    core.info(`release notes:\n${releaseNotes}`)
+
+    core.setOutput('oldversion', `${config.v}${currentVersion}`)
+    core.setOutput('version', `${config.v}${newVersion}`)
+    core.setOutput('releasenotes', releaseNotes)
```

Comment:
should there be spaces?
'old version'
'release notes'

Reply:
These are output variables to be accessed and used by other steps in a job, so they cannot contain spaces. Under the hood, inputs/outputs are just special env vars, and spaces are converted to underscores, but I prefer to be deliberate. I've updated these to use dashes to make them easier to read.